src/app/scripts/cac/control/cac-control-sidebar-directions.js

```diff
@@ -162,6 +165,9 @@ CAC.Control.SidebarDirections = (function ($, Control, BikeModeOptions, Geocoder
             $.extend(otpOptions, {optimize: 'TRIANGLE'},
                      bikeModeOptions.options.bikeTriangle[bikeTriangle]);
             UserPreferences.setPreference('bikeTriangle', bikeTriangle);
+
+            // allow longer bike riding when using public transit
+            $.extend(otpOptions, { maxWalkDistance: MAXWALK_BIKE * METERS_PER_MILE });
```

Comment:
No; if the mode switches to non-bike, the max walk preference gets updated or unset (see block below). I guess it's fine for now since we don't have a user control for changing it for bike yet.